src/main/java/uk/gov/companieshouse/model/validator/CompanyNumberValidator.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.companieshouse.model.validator;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CompanyNumberValidator implements ConstraintValidator<ValidCompanyNumber, String> {
+
+    private final Pattern prefixListRegex = Pattern.compile("(?i)^([A-Z][A-Z]?)\\b(,[A-Z][A-Z]?)*$");
+
+    private final Pattern companyNumberRegex;
+
+    public CompanyNumberValidator(@Value("${companyNumber.prefixes}") String prefixString) {
```

Comment:
what happens if this config does not exist?

Reply:
Standard behaviour. It will throw an exception

Reply:
instead of creating a constructor, why not override the `initialize` method on the ConstraintValidator?

Reply:
What's the benefit?

Reply:
well it's there for us to use exactly for this purpose, why define something new?

Reply:
it's there for when you want to create an anonymous class where you won't have access to a constructor right?

Reply:
<img width="790" alt="Screenshot 2020-06-24 at 15 47 25" src="https://user-images.githubusercontent.com/16392604/85578599-18dd5580-b632-11ea-86dc-775cda828dcf.png">

Reply:
I know your constructor is doing exactly the same, but since we are implementing this class and this method exists (guaranteed to be called), then why not use it for validating the config?